slicers/fill_layer.go

```diff
@@ -7,16 +7,71 @@ import (
 	"sort"
 )
 
-func CalcFillPlanes(mesh *Mesh, settings Settings) []Plane {
-	minx, maxx := mesh.MinMaxZ(AxisX)
+func CalcFillPlanes0(mesh *Mesh, settings Settings, MyAxis Vector, step float64) []Plane {
+	minx, maxx := mesh.MinMaxZ(MyAxis)
+	curP := MyAxis.MulScalar(0).ToPoint()
+	nMax := int(math.Ceil((maxx - 0) / step))
+	nMin := int(math.Ceil((0 - minx) / step))
+
+	planes := []Plane{}
+	planes = append(planes, Plane{curP, MyAxis})
+
+	for i := 0; i < nMax; i++ {
+		curP = curP.Shift(MyAxis.MulScalar(step))
+		planes = append(planes, Plane{curP, MyAxis})
+	}
+
+	curP = MyAxis.MulScalar(0).ToPoint()
```

Comment:
MulScalar(0) return 0, is it right behaviour?

Reply:
yes